server/routes/auth.js

```diff
@@ -16,17 +18,46 @@ router.post("/register", async (req, res, next) => {
         next(errorMessage);
       } else {
         let hash = await bcrypt.hash(req.body.password, 10);
-        let newUser = new User({
-          username: req.body.username,
-          password: hash
-        });
-        newUser.save(function(err, user) {
+        SettingsModel.findOne({ email: req.params.email }, (err, settings) => {
           if (err) {
             next(err);
           }
-          console.log("New user created!");
-          res.status(201).send({ success: true });
-        });
+          if (settings) {
+            next("Settings for this user already exist!");
+          } else {
+            let settings = new SettingsModel({
+              _id: mongoose.Types.ObjectId(),
+              email: req.body.username,
+              companies: [req.body.company],
+              platforms: {
+                reddit: true,
+                twitter: true,
+                facebook: true,
+                amazon: true,
+                forbes: true,
+                shopify: true,
+                businessInsider: true
+              }
+            });
+            settings.save((err) => {
+              if (err) {
+                next(err);
+              }
+            });
+            let newUser = new User({
+              username: req.body.username,
+              password: hash,
+              settings
+            });
+            newUser.save(function(err, user) {
+              if (err) {
+                next(err);
+              }
+              console.log("New user created!");
+              res.status(201).send({ success: true });
+            });
+          }
+        })
```

Comment:
I think the logic here would be to first create the setting object. Then, on successful save of the settings, create the user. The user wouldn't save the entire settings object, instead, should just save the setting's mongoose objectId as the setting field. Saving the entire settings object into the user defeats the purpose of having a separate setting schema. So I think it should look like: 
...
settings.save((err, setting) => {
if (err) next(err);
let newUser = new User({ username: req.body.username, password: hash, settings: setting._id});
newUser.save(error => {
if (error) next(err);
console.log("New user created");
res.status(201).send({ success: true });
});
});

Reply:
I changed it